### PR TITLE
Make casting work

### PR DESCRIPTION
**Pull request by svenallers**
Hi,
I realized during some implementation that a cast you make is actually not applied. I fixed that by overriding `visitCasting()` in SqlFormatter. During the implementation of `visitCasting()` , I also realized that
```Kotlin
if (expr.removeBrackets) {
    visit(expr)
} else {
    write("(")
    visit(expr)
    removeLastBlank()
    write(") ")
}
```
was a very common pattern used all over the place. I took the chance and refactored it into a single method `writeExpression()` to reduce the redundancy. I hope that's fine. Otherwise, I would revert it.

### CODE REVIEW

**Comment by svenallers** [comment]
resolves #146

**Comment by vincentlauvlwj** [comment]
Thanks for your contribution, the code is perfect, but we have to ensure the cast syntax wors for all supported dialects, could you please add unit tests for each of them?

File: ktorm-core/src/main/kotlin/org/ktorm/expression/SqlFormatter.kt [hunk 1]
```diff
@@ -418,6 +391,13 @@ public abstract class SqlFormatter(
         return expr
     }
 
+    override fun <T : Any> visitCasting(expr: CastingExpression<T>): CastingExpression<T> {
+        write("CAST(")
+        writeExpression(expr.expression)
+        write("AS ${expr.sqlType.typeName}) ")
+        return expr
```
Comment (vincentlauvlwj) [inline ktorm-core/src/main/kotlin/org/ktorm/expression/SqlFormatter.kt:398]: Use the `writeKeyword` function

**Comment by svenallers** [comment]
@vincentlauvlwj I added tests for each dialect. Was not as easy as I expected, because every dialect has its own restrictions as to what cast is allowed and MySql, Oracle as well as SQL Server do not allow casts from int to text...

File: ktorm-core/src/main/kotlin/org/ktorm/expression/SqlFormatter.kt (hunk 1 above)
Comment (svenallers) [reply ktorm-core/src/main/kotlin/org/ktorm/expression/SqlFormatter.kt:398]: Done

**Review by vincentlauvlwj — APPROVED** [review]

**Comment by vincentlauvlwj** [comment]
Merged, will release in next version.

**Pull request merged**